src/context/toastNotificationProvider.tsx

```diff
@@ -0,0 +1,150 @@
+// TODO: This should be upstreamed and combined with existing NotificationProvider
+// NOTE: It would be easy to accomplish this, just wrap NotificationProvider with this provider
+import {
+  NotificationAction,
+  NotificationType,
+  ValueOf,
+  failure,
+  info,
+  success,
+} from "@canonical/react-components";
+import { NotificationSeverity } from "@canonical/react-components/dist/components/Notification/Notification";
+import ToastNotifications from "components/ToastNotifications";
+import React, {
+  FC,
+  PropsWithChildren,
+  ReactNode,
+  createContext,
+  useContext,
+  useState,
+} from "react";
+
+export type ToastNotificationType = NotificationType & {
+  timestamp?: ReactNode;
```

Comment:
shouldn't this be string?
```suggestion
  timestamp?: string;
```

Reply:
yes, although the `Notification` component takes in a `ReactNode`, so I decided to just follow through with the same prop type

Reply:
I think it is fine to be more specific in a client component like this one. And string is more specific than ReactNode I guess. But then again, not too important anyhow.

Reply:
cool, will change it to string :+1: